### PR TITLE
Update 2.7 branch to use Quarkus 2.7.6

### PR DESCRIPTION
**Pull request by michalvavrik**
### Summary

[Quarkus 2.7.6.Final](https://github.com/quarkusio/quarkus/releases/tag/2.7.6.Final) has been released and branch 2.7 should use it.

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Dependency update
- [ ] Refactoring
- [ ] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by michalvavrik** [comment]
quarkus-qe#657 was cherry picked in order to fix errors in `ProxyIT`.

**Review by pjgg — CHANGES_REQUESTED** [review]
- Quarkus-cli: 
Works this test on 2.7.6.Final? https://github.com/quarkus-qe/quarkus-test-suite/blob/main/quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliExtensionsIT.java#L99

If does, could you remove `2.7.5.Final ` Disable annotation

- Life application module:
Please also update this reference
https://github.com/quarkus-qe/quarkus-test-suite/blob/main/lifecycle-application/pom.xml#L42

**Comment by michalvavrik** [comment]
> * Quarkus-cli:
>   Works this test on 2.7.6.Final? https://github.com/quarkus-qe/quarkus-test-suite/blob/main/quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliExtensionsIT.java#L99
> 
> If does, could you remove `2.7.5.Final ` Disable annotation

`QuarkusCliExtensionsIT` does not work at all with 2.7 stream of `registry.quarkus.io`: `Unable to list extensions: Failed to locate stream 2.7 in platform(s)`. However should be working with `registry.quarkus.redhat.com`.

> 
> * Life application module:
>   Please also update this reference
>   https://github.com/quarkus-qe/quarkus-test-suite/blob/main/lifecycle-application/pom.xml#L42

That was already updated, please see changed files.

**Comment by michalvavrik** [comment]
I added `registry.quarkus.redhat.com` as stream 2.7 does not work with `registry.quarkus.io` (2.7 has been removed from there)

File: quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliExtensionsIT.java [hunk 1]
```diff
@@ -29,6 +31,7 @@
 @QuarkusScenario
 @DisabledOnQuarkusVersion(version = "1\\..*", reason = "Quarkus CLI has been reworked in 2.x")
 @DisabledIfSystemProperty(named = "profile.id", matches = "native", disabledReason = "Only for JVM verification")
+@EnabledOnQuarkusVersion(version = ".*redhat.*", reason = "There is no stream 2.7 in code.quarkus.io anymore")
```
Comment (pjgg) [inline quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliExtensionsIT.java:34]: GH CI is running upstream quarkus version, and we should keep this behavior as it is.

**Review by pjgg — CHANGES_REQUESTED** [review]

File: quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliExtensionsIT.java [hunk 2]
```diff
@@ -40,6 +43,11 @@ public class QuarkusCliExtensionsIT {
     @Inject
     static QuarkusCliClient cliClient;
 
+    @BeforeAll
+    static void useRHBQRegistry() {
+        cliClient.run("registry", "add", "registry.quarkus.redhat.com");
```
Comment (pjgg) [inline quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliExtensionsIT.java:48]: Quarkus product coverage is done by Jenkins, and the environment is already setup by our jobs, so this `BeforeAll` should be removed

File: quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliExtensionsIT.java (hunk 1 above)
Comment (pjgg) [reply quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliExtensionsIT.java]: Well, I proceed with this amend

Comment (michalvavrik) [reply quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliExtensionsIT.java]: You can't run 2.7 against upstream as it won't work. Let's try it out.

File: quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliExtensionsIT.java (hunk 2 above)
Comment (michalvavrik) [reply quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliExtensionsIT.java]: with `BeforeAll`, you are able to run the tests f.e. with `mvn clean verify -pl quarkus-cli/ -Dit.test=QuarkusCliExtensionsIT -Dquarkus.platform.version=2.7.5.Final-redhat-00011 -Dquarkus-plugin.version=2.7.5.Final-redhat-00011 -Dquarkus.platform.group-id=com.redhat.quarkus.platform -Dquarkus.platform.artifct-id=quarkus-bom -Dmaven.repo.local=/home/mvavrik/rhbq/275CR1/rh-quarkus-2.7.5.GA-maven-repository/maven-repository`.

In my understanding, running Test Suite branch 2.7 with RHBQ is more than expected, so it's prerequisite to set registry and should be documented in a same fashion as other in README.

**Review by pjgg — APPROVED** [review]
LGTM

**Pull request merged**